pkg/kubectl/explain/field_lookup.go

```diff
@@ -78,10 +78,14 @@ func (f *fieldLookup) VisitKind(k *proto.Kind) {
 	if f.SaveLeafSchema(k) {
 		return
 	}
-
-	subSchema, ok := k.Fields[f.Path[0]]
+	var field string = f.Path[0]
+	if len(f.Path) > 1 && f.Path[1] == "" {
+		// validate the last field which can be accidently ended with period
+		field = field + "." + f.Path[1]
```

Comment:
Is there some added benefit by adding `f.Path[1]`? Since it's already equal to `""`, it shouldn't matter?

Reply:
If `f.Path[1]` is `""`, it means that a trailing period definitely exists. Maybe we can return an early error here instead of checking if `k.Fields[field]` exists?